### PR TITLE
Fix neighbour calc

### PR DESCRIPTION
**Pull request by Tetracarbonylnickel**
Addresses the problem that the shrink wrapped cell was too tight. Fix by adding a little layer if the shrink wrapped  cell is created.

### CODE REVIEW

File: apax/data/preprocessing.py [hunk 1]
```diff
@@ -110,9 +116,11 @@ def get_shrink_wrapped_cell(positions):
     cell_origin = rmin
     cell = np.diag(rmax - rmin)
     for idx in range(3):
-        if np.all(cell[:, idx] < 10e-3):
+        if cell[idx, idx] < 10e-1:
             cell[idx, idx] = 1.0
 
+    cell.flat[:: cell.shape[1] + 1] += 1
+
```
Comment (M-R-Schaefer) [inline apax/data/preprocessing.py:123]: what does this line do?

Comment (Tetracarbonylnickel) [reply apax/data/preprocessing.py:123]: same as `cell.diagonal()` idk why I used that.

File: apax/data/preprocessing.py [hunk 2]
```diff
@@ -110,9 +116,11 @@ def get_shrink_wrapped_cell(positions):
     cell_origin = rmin
     cell = np.diag(rmax - rmin)
     for idx in range(3):
-        if np.all(cell[:, idx] < 10e-3):
+        if cell[idx, idx] < 10e-1:
             cell[idx, idx] = 1.0
 
+    cell.flat[:: cell.shape[1] + 1] += 1
```
Comment (M-R-Schaefer) [inline apax/data/preprocessing.py:122]: this could be replaced by something clearer

**Review by M-R-Schaefer — CHANGES_REQUESTED** [review]

File: apax/data/preprocessing.py [hunk 3]
```diff
@@ -119,7 +119,7 @@ def get_shrink_wrapped_cell(positions):
         if cell[idx, idx] < 10e-1:
             cell[idx, idx] = 1.0
 
-    cell.flat[:: cell.shape[1] + 1] += 1
+    cell[np.diag_indices_from(cell)] += 1
```
Comment (M-R-Schaefer) [inline apax/data/preprocessing.py:122]: so the purpose of this line is to increase the cell by 1 Angstrom in all directions.
is this necessary?

Comment (Tetracarbonylnickel) [reply apax/data/preprocessing.py:122]: Yes. If the cell is too tight some atoms will be counted as being in a adjacent cell and the neighbour list will be wrong.

**Review by M-R-Schaefer — APPROVED** [review]

**Pull request merged**